### ISSUE TITLE
Redshift distribution and sort key - default to auto or allow top level configuration

### ISSUE THREAD

**Issue by brangisom** [dbt-labs/dbt#3052]
### Describe the feature
Redshift now has an `auto` option for table distribution and sort keys. Additionally, they have also released a feature, [Automatic Table Optimization](https://docs.aws.amazon.com/redshift/latest/dg/t_Creating_tables.html), that allows Redshift to swap things out. This feature more or less automates the suggestions that show up in the `svv_alter_table_recommendations` system view, UI, etc. 

It would be super helpful to have that as the default, either by convention in DBT, or by configuration through allowing the dist ~~or sort~~ configuration values to be set in the `dbt_project.yml`. 

EDIT: Looks like sort keys can be set in `dbt_project.yml` already.

### Describe alternatives you've considered
A config() block in every model. 😢 

### Additional context
This is a Redshift database specific feature.

### Who will this benefit?
Anyone who uses Redshift, and would like to default to `auto` dist and/or sort keys to take advantage of the Automatic Table Optimization feature.

### Are you interested in contributing this feature?
Sure.

**Comment by jtcohen6** [comment]
@brangisom This should already be possible in **dbt_project.yml** for both sort and dist keys:
```yml
models:
  +sort: auto
  +dist: auto
```

What I also hear you proposing, and I think it has merit, is changing the default values of Redshift-specific configs:

https://github.com/fishtown-analytics/dbt/blob/2b48152da66dbd7f07272983bbc261f1b6924f20/plugins/redshift/dbt/adapters/redshift/impl.py#L12-L16

We might change that to:
```python
class RedshiftConfig(AdapterConfig):
    sort_type: Optional[str] = None
    dist: Optional[str] = 'auto'
    sort: Optional[str] = 'auto'
    bind: Optional[bool] = false
```

Effectively, these were hard-won features that were a lot trickier a few years ago. Nowadays, we set `sort: auto`, `dist: auto`, and `bind: false` as top-level configs in every one of our dbt + Redshift projects. Of course, specific tables sometimes benefit from opinionated sorting/distributing, but that's what `config()` is for.

**Comment by brangisom** [comment]
Yeah, it turns out the auto can't be passed into a CTAS, so I confused that for the config not working. I fiddled with some things, got some errors, and then checked the docs and submitted a ticket with aws to confirm.

We can go ahead and close this out, haha

**Comment by jtcohen6** [comment]
Ok! Feel free to comment back here if you figure out something that dbt could be doing better.

**Comment by matthewrj** [comment]
Using 
```
models:
  +sort: auto
  +dist: auto
```
produces `sortkey(auto)` instead of `sortkey auto`. This is because the sort key property is treated as a column name, not an option.

I'd imagine you'd need a new config option `sortkey_auto: true/false` to support generating `SORTKEY AUTO` in the CTAS statement.

**Comment by jtcohen6** [comment]
@matthewrj Ah, fair point! Thanks for finding this issue. I think a simpler solution would involve adjusting the `sort` macro so that, if `'sort == 'auto'`, the macro returns `sortkey auto` instead of `sort auto`.

https://github.com/dbt-labs/dbt/blob/2b48152da66dbd7f07272983bbc261f1b6924f20/plugins/redshift/dbt/include/redshift/macros/adapters.sql#L17-L29

I'd welcome a contribution for that, so I'll re-open the issue.

**Comment by brangisom** [comment]
It looks like setting `sortkey auto` and `diststyle auto` is still not supported in a CTAS, though. 

See: https://docs.aws.amazon.com/redshift/latest/dg/r_CREATE_TABLE_AS.html

This change would require an `ALTER TABLE table ALTER DISTSTYLE AUTO, ALTER SORTKEY AUTO` as a dynamic post-hook of some sort, or something.
